animation_nodes/data_structures/attributes/attribute.pyx

```diff
@@ -0,0 +1,86 @@
+from .. lists.base_lists cimport (
+    LongList,
+    FloatList,
+    ColorList,
+    BooleanList,
+    Vector2DList,
+    Vector3DList,
+)
+
+cdef class Attribute:
+    def __cinit__(self, str name = None,
+                        AttributeType type = CUSTOM,
+                        AttributeDomain domain = POINT,
+                        AttributeDataType dataType = FLOAT,
+                        data = None):
+
+        self.name = name
+        self.type = type
+        self.domain = domain
+        self.dataType = dataType
+        self.data = data
+
+    def copy(self):
+        return Attribute(self.name, self.type, self.domain, self.dataType,
+                         self.data.copy())
+
+    def extend(self, data):
+        self.data.extend(data)
+
+    def repeated(self, Py_ssize_t amount = -1):
+        return Attribute(self.name, self.type, self.domain, self.dataType,
+                         self.data.repeated(amount = amount))
+
+    def appendAttribute(self, sourceAttribute = None, Py_ssize_t amount = -1):
+        if sourceAttribute is not None:
+            self.extend(sourceAttribute.data)
+        else:
+            extension = self.getDataType()(length = amount)
+            extension.fill(0)
+            self.extend(extension)
+
+    def getTypeAsString(self):
```

Comment:
I still don't see why we need all of those string functions. Why not just use the enums we defined?

Reply:
Omar, enums returns `int` not string. So. we have to map them manually.